### DISCUSSION TITLE
是否可以把waline的用户登录系统集成到博客中？

### DISCUSSION THREAD

**Discussion by scjmjy** [walinejs/waline#1557] [Q&A] [answered]
我的博客想要有用户系统，支持用户登录，该怎么使用waline现有的用户系统，集成到我的博客里呢？

**Comment by lizheming** [comment, marked as answer]
可以的，可以参考这个 DEMO 实现一下；里面包含了获取用户信息、登录、退出登录三个登录相关操作。https://codesandbox.io/s/youthful-sea-bc6czb?file=/index.html:0-3024

**Reply by scjmjy** [reply]
多谢！
可以实现基本的功能。
不过复杂一点的，现在还不太好实现吧：比如联动，如何得知评论组件点击了logout？博客登录之后如何通知评论组件更新登录状态？

**Reply by lizheming** [reply]
1. 可以通过监听 localStorage 的 变化得知 login 和 logout 的回调 https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event
2. 登录之后可以直接调用 [update](https://waline.js.org/guide/client/spa.html#update) 方法更新组件
3. 如果觉得这些都不太好，也可以自己写 Waline 的客户端适配你的场景

**Comment by scjmjy** [comment]
好的，谢谢耐心回复☕
